### PR TITLE
Confirm subscription

### PR DESCRIPTION
**Pull request by gbirke**
First steps towards the ["Confirm Subscription" use case](https://phabricator.wikimedia.org/T123045).

### CODE REVIEW

File: tests/Integration/Presentation/Presenters/ConfirmSubscriptionHtmlPresenterTest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class ConfirmSubscriptionHtmlPresenterTest extends \PHPUnit_Framework_TestCase {
+
+	public function testGivenSuccessResponse_templateIsRenderedWithoutMessages() {
+		$twig = $this->getMockBuilder( TwigTemplate::class )->disableOriginalConstructor()->getMock();
+		$twig->expects( $this->once() )
+			->method( 'render' )
+			->with( [] );
+		$presenter = new ConfirmSubscriptionHtmlPresenter( $twig );
+		$presenter->present( ValidationResponse::newSuccessResponse() );
+	}
```
Comment (JeroenDeDauw) [inline tests/Integration/Presentation/Presenters/ConfirmSubscriptionHtmlPresenterTest.php]: The way you wrote those tests, they are unit tests no?

What I did so far is use real twig and see certain HTML is present in the output. Not saying that approach is better, just pointing out a potential interesting difference.

File: tests/System/Routes/ConfirmSubscriptionRouteTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+use WMDE\Fundraising\Frontend\Messenger;
+use Swift_NullTransport;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class ConfirmSubscriptionRouteTest extends WebRouteTestCase {
+
+	// @codingStandardsIgnoreStart
```
Comment (JeroenDeDauw) [inline tests/System/Routes/ConfirmSubscriptionRouteTest.php]: I think we should just disable this thing. Too many false positives. And our IDEs highlight those occurrences anyway (with less false positives than PHPCS).

File: tests/System/Routes/ConfirmSubscriptionRouteTest.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	protected function onTestEnvironmentCreated( FunFunFactory $factory, array $config ) {
+		// @codingStandardsIgnoreEnd
+		$factory->setMessenger( new Messenger(
+			Swift_NullTransport::newInstance(),
+			$factory->getOperatorAddress()
+		) );
+	}
+
+	public function testGivenAnUnconfirmedSubscriptionRequest_successPageIsDisplayed() {
```
Comment (JeroenDeDauw) [inline tests/System/Routes/ConfirmSubscriptionRouteTest.php]: If you have this, then I'd also expect to have a `testGivenAConfirmedSubscriptionRequest_stuffHappens`. Or does that not make sense?

File: tests/System/Routes/ConfirmSubscriptionRouteTest.php (hunk 2 above)
Comment (gbirke) [inline tests/System/Routes/ConfirmSubscriptionRouteTest.php]: Go ahead.

**Pull request merged**